plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java

```diff
@@ -97,8 +97,8 @@ public DeltaLakeWriter(
         this.fileWriter = requireNonNull(fileWriter, "fileWriter is null");
         this.rootTableLocation = requireNonNull(rootTableLocation, "rootTableLocation is null");
         this.relativeFilePath = requireNonNull(relativeFilePath, "relativeFilePath is null");
-        this.partitionValues = partitionValues;
-        this.stats = stats;
+        this.partitionValues = requireNonNull(partitionValues, "partitionValues is null");
```

Comment:
We usually add `ImmutableList.copyOf` instead:
```suggestion
        this.partitionValues = ImmutableList.copyOf(partitionValues);
```

Reply:
There will be null value in the `partitionValues`

Reply:
Let's add a code comment, then.